pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -1142,6 +1149,7 @@ cdef class StringHashTable(HashTable):
                 # ignore_na is True), we can skip the actual value, and
                 # replace the label with na_sentinel directly
                 labels[i] = na_sentinel
+                seen_na = True
```

Comment:
Instead of trying to find this iterating by element can we get the same performance by querying up front for missing values? That approach would work better in the future where we are more arrow based

Reply:
Arrow  strings are using a completely different code path, this is just for our own strings

Reply:
> wouldn't we expect a two-pass implementation to be slower? i

Generally I would expect a two pass solution to be faster. Our nulll-checking implementations are pretty naive and always go elementwise with a boolean value. The Arrow implementation iterates 64 bits at a time, so checking for NA can be up to 64x as fast as this kind of loop. I'm not sure what NumPy does internally but with a byte mask that same type of logic would be up to 8x as fast.

By separating that out to a different `has_na().any()` type of check you let other libraries determine that much faster than we can

Reply:
To be clear not a blocker for now; just something to think about as we make our code base more Arrow friendly